deploy/stacks/backend_stack.py

```diff
@@ -65,6 +66,8 @@ def __init__(
         vpc = self.vpc_stack.vpc
         vpc_endpoints_sg = self.vpc_stack.vpce_security_group
 
+        self.pivot_role_name = f"dataallPivotRole{'-cdk' if enable_pivot_role_auto_create else ''}"
```

Comment:
During a migration to new auto-create pivot roles, there will be a point in time when backend has updated and is configured to use the new pivot roles (with `-cdk` suffix), while environment updates are still pending. As I understand, the IAM policies will change allowing only new pivot roles to perform actions. Is it expected that environments won't be usable until they upgrade? 

Also, I'm not as familiar with the codebase but we need to be extra cautious here to avoid being locked out. Pivot role isn't itself used to perform the deployment right?

Reply:
I added a "capital letter" warning in the description and we will make sure to add it in the release notes, but thank you you are right there is some downtime until everything is complete.